classpath/java/lang/Integer.java

```diff
@@ -139,4 +139,14 @@ public static Integer decode(String string) {
     }
     return new Integer(parseInt(string, 10));
   }
+
+  public static int numberOfLeadingZeros(int i) {
```

Comment:
I don't know about the right shift, but it seems that it shouldn't be an unsigned one. I agree that adding test cases would be a good idea, so i added some.